lib/db/util.js

```diff
@@ -84,6 +85,19 @@ module.exports = {
     }
   },
 
+  buffersAreEqual(buffer1, buffer2) {
+    buffer1 = Buffer.from(buffer1, 'hex')
+    buffer2 = Buffer.from(buffer2, 'hex')
+    let mismatch = buffer1.length - buffer2.length
+    if (mismatch) {
+      return false
+    }
+    for (let i = 0; i < buffer1.length; i++) {
+      mismatch |= buffer1[i] ^ buffer2[i]
+    }
+    return mismatch === 0
```

Comment:
Is the reason for using this function instead of `buffer1.equals(buffer2)` so that the comparison occurs in constant time? If so maybe we should make that clear in the function name, e.g. the auth server has a `bufferEqualConstantTime` IIRC.

(actually, just checked and the auth server uses a dep instead, but still worth naming this explicitly I think)

Reply:
Turns out that this was added to [Node 6.6](https://nodejs.org/api/crypto.html#crypto_crypto_timingsafeequal_a_b). Updated to use that instead.

Reply:
Oh, nice!